Hi @kaste, thanks for the quick response.

I could remove the multiple-files commit from the PR and submit a new PR later if the main plugin's changes don't fix it automatically. In the meantime I'll use the fix locally.

Here's some typical output from `ghc`:

```
mean.hs:20:1: warning: [-Wmissing-signatures]
    Top-level binding with no type signature:
      scanl' :: Traversable t1 => (a -> t2 -> a) -> a -> t1 t2 -> t1 a
   |
20 | scanl' f = (snd .) . mapAccumL (\acc x -> (f acc x, f acc x))
   | ^^^^^^
```

Did you want me to include it in the code somewhere?

I didn't understand your last remark ("Have you considered …") because some text is missing. Could you explain your idea again, please?

--- a/linter.py
+++ b/linter.py
@@ -25,6 +25,11 @@ class Ghc(Linter):
     # ghc writes errors to STDERR
     error_stream = util.STREAM_STDERR
 
+    def finalize_cmd(self, cmd, context, **kwargs):
+        """Override to capture temporary filename."""
+        self.temp_file = context['temp_file']
+        return super().finalize_cmd(cmd, context, **kwargs)
+
     def split_match(self, match):
         """Override to ignore errors reported in imported files."""
         match, line, col, error, warning, message, near = (
@@ -32,7 +37,7 @@ class Ghc(Linter):
         )
 
         match_filename = basename(match.groupdict()['filename'])
-        linted_filename = basename(self.filename)
+        linted_filename = basename(self.temp_file)
 
         if match_filename != linted_filename:
             return None, None, None, None, None, '', None

--- a/linter.py
+++ b/linter.py
@@ -6,9 +6,11 @@ class Ghc(Linter):
     cmd = ('ghc', '-fno-code', '-Wall', '-Wwarn', '-fno-helpful-errors',
            '$temp_file')
     regex = (
-        r'^(?P<filename>.+):'
-        r'(?P<line>\d+):(?P<col>\d+):'
-        r'\s+(?P<warning>Warning:\s+)?(?P<message>.+)$'
+        r'\s*(?P<filename>.+):'
+        r'\s*(?P<line>\d+):(?P<col>\d+):'
+        r'\s*(?:(?P<warning>[Ww]arning):|(?P<error>[Ee]rror):)?'
+        r'\s*(?P<flag>\[-W[^\]]*\])?'
+        r'\s*(?P<message>.+?):?$'
     )
     multiline = True
     defaults = {
@@ -41,5 +43,8 @@ class Ghc(Linter):
 
         if match_filename != linted_filename:
             return None, None, None, None, None, '', None
+
+        if match.groupdict()['flag']:
+            message += " " + match.groupdict()['flag']
 
         return match, line, col, error, warning, message, near

--- a/linter.py
+++ b/linter.py
@@ -3,7 +3,8 @@ from os.path import basename
 
 
 class Ghc(Linter):
-    cmd = ('ghc', '-fno-code', '-Wall', '-Wwarn', '-fno-helpful-errors')
+    cmd = ('ghc', '-fno-code', '-Wall', '-Wwarn', '-fno-helpful-errors',
+           '$temp_file')
     regex = (
         r'^(?P<filename>.+):'
         r'(?P<line>\d+):(?P<col>\d+):'

